networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -253,11 +267,11 @@ SafeFuture<Optional<BlobSidecar>> loadNextBlobSidecar() {
 
     boolean isComplete() {
       return currentSlot.get().isGreaterThan(maxSlot)
-          || maxRequestBlobSidecars.isLessThanOrEqualTo(sentBlobSidecars.get());
+          || UInt64.valueOf(sentBlobSidecars.get()).isGreaterThanOrEqualTo(maxRequestBlobSidecars);
     }
 
     void incrementSlotAndIndex() {
-      if (currentIndex.get().equals(maxBlobsPerBlock.minus(UInt64.ONE))) {
+      if (currentIndex.get().equals(UInt64.valueOf(blobSidecarsCount).minusMinZero(UInt64.ONE))) {
```

Comment:
I'd put `currentIndex.get().equals(UInt64.valueOf(blobSidecarsCount).minusMinZero(UInt64.ONE))` to some private method, it's used twice and it could have some speaking name

Reply:
Done